### ISSUE TITLE
docs: types need reference documentation

### ISSUE THREAD

**Issue by jalehman** [urbit/shrub#68]
There are many types introduced in neo that are either new or just heavily used: `kook, curb, stud, kids, dare, lads, pish, lash, form, pail, note` to name a few. There should be reference documentation for these in one place.

**Comment by tiller-tolbus** [comment]
There are some arms in sur/neo related to axals in the `+of` core that would be very helpful to document for purposes of writing renderers like Sky and Tree

**Comment by bonbud-macryg** [comment]
@tiller-tolbus Yeah I've scanned /sur/neo again and I'm thinking there's a lot of functionality in there that should be covered in a separate document. Maybe one doc for `+of` and one for using the shanes.

**Comment by bonbud-macryg** [comment]
@liam-fitzgerald When you're back could you go through `types.txt` in #80 and leave comments on my questions in there? If you CMD+F `???` or `[` that should get all of them. Once we've filled those out the doc should cover 1) all of the types shrub devs have to think about outside of `+of` and shanes and 2) the types that *those* types are made up of.

**Comment by bonbud-macryg** [comment]
Would be nice to have anchor links in docs like this but Sky doesn't support it rn.
